pkg/cmd/release-payload-controller/job_state_controller.go

```diff
@@ -0,0 +1,220 @@
+
+	// If there are no results, then there is nothing to do...
+	if attempts == 0 {
+		return v1alpha1.JobStateUnknown
+	}
+
+	switch {
+	case job.AnalysisJobCount > 0:
+		return processAnalysisJobResults(job.JobRunResults, job.AnalysisJobCount)
+	case job.MaxRetries > 0:
+		return processRetryJobResults(job.JobRunResults, job.MaxRetries)
+	default:
+		return processJobResults(job.JobRunResults)
+	}
+}
+
+// interrogateJobResults organizes JobRunResults into categorized slices for further processing.  The individual jobs
+// are sorted into the following categories: Pending, Successful, Failed, and Unknown.
+func interrogateJobResults(results []v1alpha1.JobRunResult) (pendingJobs, successfulJobs, failedJobs, unknownJobs []v1alpha1.JobRunResult) {
+	for _, result := range results {
+		switch result.State {
+		case v1alpha1.JobRunStatePending, v1alpha1.JobRunStateTriggered:
+			pendingJobs = append(pendingJobs, result)
+		case v1alpha1.JobRunStateSuccess:
+			successfulJobs = append(successfulJobs, result)
+		case v1alpha1.JobRunStateFailure, v1alpha1.JobRunStateAborted, v1alpha1.JobRunStateError:
+			failedJobs = append(failedJobs, result)
+		default:
+			unknownJobs = append(unknownJobs, result)
+		}
+	}
+
+	return pendingJobs, successfulJobs, failedJobs, unknownJobs
+}
+
+// processJobResults processes JobRunResults for jobs that are not configured as analysis jobs or
+// that have retries.  Any Successful result of this class of jobs returns a "Success" JobState.
+func processJobResults(results []v1alpha1.JobRunResult) v1alpha1.JobState {
+	var pendingJobs, successfulJobs, failedJobs, _ = interrogateJobResults(results)
+
+	switch {
+	case len(successfulJobs) > 0:
+		return v1alpha1.JobStateSuccess
+	case len(pendingJobs) > 0:
+		return v1alpha1.JobStatePending
+	case len(failedJobs) > 0:
+		return v1alpha1.JobStateFailure
+	default:
+		return v1alpha1.JobStateUnknown
+	}
+}
+
+const (
+	// analysisSuccessThreshold represents the threshold that must be reached or exceeded for the
+	// corresponding analysis jobs to be marked successful
+	analysisSuccessThreshold = 0.6
+)
+
+// processAnalysisJobResults processes JobRunResults for analysis jobs.  The JobState will be
+// "Unknown" until the number of JobRunResults equals count.  While any of the JobRunResults are
+// still running (pending), the JobState will be "Pending".  Once all the JobRunResults are complete,
+// a success rate, that is greater-than or equal to analysisSuccessThreshold, will return a "Success"
+// JobState.  Anything less-than analysisSuccessThreshold will return a "Failure" JobState.
+func processAnalysisJobResults(results []v1alpha1.JobRunResult, count int) v1alpha1.JobState {
+	var pendingJobs, successfulJobs, _, _ = interrogateJobResults(results)
+
+	switch {
+	case len(results) < count:
+		return v1alpha1.JobStateUnknown
+	case (float64(len(successfulJobs)) / float64(count)) >= analysisSuccessThreshold:
+		return v1alpha1.JobStateSuccess
+	case len(pendingJobs) > 0:
```

Comment:
If it mathematically impossible to get enough passes with the remaining jobs you could fail